src/oneD/Sim1D.cpp

```diff
@@ -511,8 +511,34 @@ int Sim1D::setFixedTemperature(doublereal t)
     return np;
 }
 
-void Sim1D::setRefineCriteria(int dom, doublereal ratio,
-                              doublereal slope, doublereal curve, doublereal prune)
+double Sim1D::fixedTemperature()
+{
+    double t_fixed = -1.;
+    for (size_t n = 0; n < nDomains(); n++) {
+        StFlow* d = dynamic_cast<StFlow*>(&domain(n));
+        if (d && d->domainType() == cFreeFlow && d->m_tfixed > 0) {
+            t_fixed = d->m_tfixed;
+            break;
+        }
+    }
+    return t_fixed;
+}
+
+double Sim1D::fixedTemperatureLocation()
+{
+    double z_fixed = -1.;
```

Comment:
Is -1 the best value to return if no fixed temperature location has been set? Should this return NaN instead? Or raise an exception? There's a similar question for the temperature, but at least -1 is clearly invalid there.

Reply:
opting for NaN without raising an exception.